Allow essential syscalls for File System API in WebContent process sandbox

szewai opened the pull request:
#### d186d78bb19920b48544fff3dee1b7d8b71a2fad
<pre>
Allow essential syscalls for File System API in WebContent process sandbox
<a href="https://bugs.webkit.org/show_bug.cgi?id=267096">https://bugs.webkit.org/show_bug.cgi?id=267096</a>
<a href="https://rdar.apple.com/120439594">rdar://120439594</a>

Reviewed by Per Arne Vollan.

269812@main blocked these syscalls and File System tests started to fail.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/272698@main">https://commits.webkit.org/272698@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19b3a4c179c7ecf1c526c8922ef46600a4ab6c3f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32583 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11335 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34435 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35153 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29456 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13683 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8527 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28973 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33021 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9553 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29114 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8329 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8464 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36489 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29616 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29479 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34585 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8592 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6546 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32446 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10268 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7604 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9202 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9187 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1725,6 +1725,8 @@
 (define (syscall-unix-in-use-after-launch)
     (syscall-number
+        SYS_ftruncate
+        SYS_fsync
```
Is it feasible to perform these file operations in e.g. the Networking process? Long term, I think it would be good to be able to block file operations in the WebContent process.

pvollan replied:
If it is substantial work to avoid these syscalls in the WebContent process, I think it is OK to add them back.

szewai replied:
> Is it feasible to perform these file operations in e.g. the Networking process? Long term, I think it would be good to be able to block file operations in the WebContent process.

The API is created for performant file access (https://fs.spec.whatwg.org/#api-filesystemfilehandle-createsyncaccesshandle), so the methods are sync (https://fs.spec.whatwg.org/#api-filesystemsyncaccesshandle) and the operations are performed in web process directly. 
Moving to network process means web process needs to be blocked waiting network process's result, which would defeat the goal.

pvollan replied:
That makes sense, R=me.

pvollan reviewed (approved):
R=me

szewai committed:
Allow essential syscalls for File System API in WebContent process sandbox
https://bugs.webkit.org/show_bug.cgi?id=267096
rdar://120439594

Reviewed by Per Arne Vollan.

269812@main blocked these syscalls and File System tests started to fail.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/272698@main
